### PR TITLE
Load plugins to vscode java extension.

### PR DESCRIPTION
**Pull request by yaohaizh**
Signed-off-by: Yaohai Zheng <yaozheng@microsoft.com>

@fbricon @gorkem @aeschli 

https://github.com/redhat-developer/vscode-java/issues/282

### CODE REVIEW

**Comment by aeschli** [comment]
Given that the new 'bundles install' request is only needed a single time (on initialization), maybe it's better to just pass the list of bundles to add to the language on initialize.
That keeps the APIs as simple as possible.

The [LSP InitializeRequest](https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md#initialize-request) has a property `initializationOptions` that can be used for that.

**Comment by yaohaizh** [comment]
If we go through  the LSP InitializeRequest , should we still expose the installbundle/uninstallbundle protocol for 3rd-party to load/unload themselves dynamically in time?

**Comment by aeschli** [comment]
> should we still expose the installbundle/uninstallbundle protocol for 3rd-party to load/unload themselves dynamically in time?

I don't think there's a need for it, so I'd suggest to remove it. Its also hard to imagine how uninstalling at runtime will work if that bundle is already in use.

**Comment by yaohaizh** [comment]
Thanks, Martin. Updated the code and this did make the code logic clear and clean.

**Comment by gorkem** [comment]
I think I am comfortable with this solution and implementation looks good. @yaohaizh can you possibly add tests for it, and we can then merge?

**Comment by yaohaizh** [comment]
@gorkem You mean test case on the jdt.ls side or the vscode java side? For the jdt.ls side, i have add more testing for loading extension. If for vscode-java side, i might need refactor current code for testing private methods

**Comment by gorkem** [comment]
@yaohaizh client side, something that massages the `javaExtension` contribution logic.

File: src/plugin.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+import { Extension } from 'vscode';
+import * as path from 'path';
+
+export function collectionExtensions(extensions: Extension<any>[]): string[] {
```
Comment (fbricon) [inline src/plugin.ts]: please rename to `collectJavaExtensions`

**Comment by fbricon** [comment]
Apart from the function name I'd like to see renamed, it will also a require documentation on contributing Java extensions, in https://github.com/redhat-developer/vscode-java/wiki, that we can reference from a paragraph in  the README.md.

**Comment by yaohaizh** [comment]
Add the wiki for extension side, yet need add server side wiki but I don't have the permission yet to add eclipse/eclipse.jdt.ls wiki.

File: test/extension.test.ts [hunk 2]
```diff
@@ -62,4 +65,20 @@ suite('Java Language Extension', () => {
 		assert.equal(vmArgs[2], '-Dfoo=Some "crazy" stuff');
 	});
 
+	test('should collection java extensions', function () {
```
Comment (akaroml) [inline test/extension.test.ts]: collection -> collect

**Pull request merged**